creator-node/src/redis.js

```diff
@@ -1,36 +1,123 @@
+/**
+ * Exports a Singleton Redis client instance, with custom wallet write locking logic
+ */
+
+const Redis = require('ioredis')
+
 const config = require('./config.js')
 const { logger: genericLogger } = require('./logging')
-const Redis = require('ioredis')
+const { asyncRetry } = require('./utils')
 
 const redisClient = new Redis(config.get('redisPort'), config.get('redisHost'))
 
-const EXPIRATION = 60 * 60 * 2 // 2 hours in seconds
-class RedisLock {
-  static async setLock(key, expiration = EXPIRATION) {
-    genericLogger.info(`SETTING LOCK ${key}`)
-    // set allows you to set an optional expire param
-    return redisClient.set(key, true, 'EX', expiration)
-  }
+const WalletWriteLock = {
+  WALLET_WRITE_LOCK_EXPIRATION_SEC: 1800, // 30 min in sec
 
-  static async getLock(key) {
-    genericLogger.info(`GETTING LOCK ${key}`)
-    return redisClient.get(key)
-  }
+  VALID_ACQUIRERS: {
+    UserWrite: 'userWrite',
```

Comment:
this isn't used currently right?

Reply:
if we decide to keep UserWrite we should probably have a way to disambiguate different locks from the key name. otherwise the key name would collide

Reply:
fair enough - i'll remove the UserWrite case for now